packages/Search/test/tstLinearBVH.cpp

```diff
@@ -86,6 +87,73 @@ TEUCHOS_UNIT_TEST_TEMPLATE_1_DECL( LinearBVH, bounds, DeviceType )
     TEST_EQUALITY( bounds[5], 6.0 );
 }
 
+TEUCHOS_UNIT_TEST_TEMPLATE_1_DECL( LinearBVH, queries, DeviceType )
+{
+    Kokkos::View<DataTransferKit::Box *, DeviceType> boxes( "boxes", 2 );
+    auto boxes_host = Kokkos::create_mirror_view( boxes );
+    boxes_host( 0 ) = DataTransferKit::Box( {0., 0., 0., 0., 0., 0.} );
+    boxes_host( 1 ) = DataTransferKit::Box( {1., 1., 1., 1., 1., 1.} );
+    Kokkos::deep_copy( boxes, boxes_host );
+    DataTransferKit::BVH<DeviceType> bvh( boxes );
+
+    // `out` and `successs` need to be captured by reference  They come from the
+    // test assertion macros expansion.
+    auto check_results = [&bvh, &out, &success](
+        std::vector<DataTransferKit::Box> const &overlap_boxes,
+        std::vector<int> const &indices_ref,
+        std::vector<int> const &offset_ref ) {
+        Kokkos::View<DataTransferKit::Details::Overlap *, DeviceType> queries(
+            "queries", overlap_boxes.size() );
+        auto queries_host = Kokkos::create_mirror_view( queries );
+        for ( int i = 0; i < queries.extent_int( 0 ); ++i )
+            queries_host( i ) =
+                DataTransferKit::Details::Overlap( overlap_boxes[i] );
+        Kokkos::deep_copy( queries, queries_host );
+
+        Kokkos::View<int *, DeviceType> indices( "indices" );
+        Kokkos::View<int *, DeviceType> offset( "offset" );
+        bvh.query( queries, indices, offset );
+
+        // Desperate attempt to reduce boiler plate code...
+        auto kokkos_view_to_teuchos_array = [](
+            Kokkos::View<int *, DeviceType> view ) {
+            auto view_host = Kokkos::create_mirror_view( view );
+            deep_copy( view_host, view );
+            Teuchos::ArrayView<int> tmp( view_host.data(), view.extent( 0 ) );
```

Comment:
OK I removed it.  I went ahead and squashed the changes and then forced the push.